doc/sphinx/proofs/writing-proofs/proof-mode.rst

```diff
@@ -1,66 +1,165 @@
-of the goal. Initially, the local context contains the local variables and
-hypotheses of the current section (see Section :ref:`gallina-assumptions`) and
-the local variables and hypotheses of the theorem statement. It is enriched by
-the use of certain tactics (see e.g. :tacn:`intro`).
+.. todo PR: my example is pretty bad--can you suggest a better one?
+   Also need to position/integrate this better
 
-When a proof is completed, the message ``Proof completed`` is displayed.
-One can then register this proof as a defined constant in the
-environment. Because there exists a correspondence between proofs and
-terms of λ-calculus, known as the *Curry-Howard isomorphism*
-:cite:`How80,Bar81,Gir89,H89`, Coq stores proofs as terms of |Cic|. Those
-terms are called *proof terms*.
+The :gdef:`proof state`, also referred to as the :gdef:`proof context`, consists of one
+or more unproven goals.
+Each goal has a :gdef:`conclusion` (the statement that is to be proven)
+and may have zero or more named facts that can be used in proving the
+conclusion (in addition to definitions, proven theorems, etc.).
 
+The term ":gdef:`goal`" may refer to an entire goal or to the conclusion
+of a goal, depending on the context.
 
-.. exn:: No focused proof.
+The facts appear above a line and the conclusion appears below that line.
+They are both types.  Facts can be :gdef:`hypotheses <hypothesis>`
+(which are propositions), variables or local definitions.
+The name of the fact appears first.  The type is shown after the colon.  Local
+definitions are shown in the form `n := 0 : nat`.
 
-   Coq raises this error message when one attempts to use a proof editing command
-   out of the proof editing mode.
+The proof state when you begin proving a theorem shows
+the statement of the theorem below the line and no hypotheses above the
+line:
+
+.. coqtop:: none
+
+   Parameter P: nat -> Prop.
+
+.. coqtop:: out
+
+   Goal forall n m: nat, n > m -> P 1 /\ P 2.
+
+After applying the :tacn:`intros` :term:`tactic`, we see hypotheses above the line.
+The names of variables (`n` and `m`) and hypotheses (`H`) appear before a colon, followed by
+the type they represent.
+
+.. coqtop:: all
+
+   intros.
+
+Some tactics, such as :tacn:`split`, create new goals.
+Goals are numbered from 1 to N at each step of the proof to permit applying a
+tactic to specific goals.
+
+.. coqtop:: all
+
+   split.
+
+"Variables" may refer specifically to items for which "type of (type of *fact*)"
+is `Set` or `Type`, and "hypotheses" to :term:`propositions <proposition>`,
+which are items for which "type of (type of
+*fact*)" is `Prop` or `SProp`, but these terms are also used interchangeably.
+
+.. coqtop:: all
+
+   let t_n := type of n in idtac "type of n:" t_n;
+   let tt_n := type of t_n in idtac "type of (type of n):" tt_n.
+   let t_H := type of H in idtac "type of H:" t_H;
+   let tt_H := type of t_H in idtac "type of (type of H):" tt_H.
```

Comment:
Let's save the reader from seeing such Ltac code so early by only displaying the output. Also, another suggestion that eases the reading IMHO:

```suggestion
.. coqtop:: out

   let t_n := type of n in idtac "type of n:" t_n;
   let tt_n := type of t_n in idtac "type of" t_n ":" tt_n.
   let t_H := type of H in idtac "type of H:" t_H;
   let tt_H := type of t_H in idtac "type of" t_H ":" tt_H.
```

Reply:
I considered doing it both ways.  Leaving out the code may leave a question unanswered.  I suppose the code could be in a footnote.  I made it "out" only.